### ISSUE TITLE
[BUG] unknown statistics error, found when storing data on backend; classifier: bayes

### ISSUE THREAD

**Issue by urmel1960** [rspamd/rspamd#3339]
### Prerequisites

* [ X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**
I do now get the following errors in rspamd-log

`2020-04-16 14:37:56 #678(normal) <6f0241>; task; rspamd_stat_backends_learn: no runtime for backend redis; classifier bayes; symbol BAYES_SPAM
2020-04-16 14:37:56 #678(normal) <6f0241>; task; rspamd_stat_backends_learn: no runtime for backend redis; classifier bayes; symbol BAYES_HAM
2020-04-16 14:37:56 #678(normal) <6f0241>; task; rspamd_task_process: learn error: Unknown statistics error, found when storing data on backend; classifier: bayes`

### Steps to Reproduce

This happens for all mails which are accepted (no action) by rspamd

**Expected behavior**
error does not occcur any longer

### Versions

Rspamd daemon version 2.5
Linux 4.15.0-96-generic #97-Ubuntu SMP Wed Apr 1 03:25:46 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### Additional Information

If there are more infos necessary - I am always happ to provide those - please let me know Thanks

**Comment by dellarte** [comment]
Same behavior here. With rspamd 2.5

**Comment by Orgoth** [comment]
Same behavior here. With rspamd 2.5

**Comment by MiCRoPhoBIC** [comment]
Same here Debian Buster with the latest 2.5 packages from rspamd repository (not from Debian). 

 #30772(controller) <390082>; csession; rspamd_task_process: learn error: Unknown statistics error, found when storing data on backend; classifier: (null)

**Comment by mans39** [comment]
Same behavior here too. With rspamd 2.5.

Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster

redis-server: *** 5:5.0.3-4+deb10u1 500

**Comment by pvanthony** [comment]
> Same behavior here too. With rspamd 2.5.
> 
> Distributor ID: Debian
> Description: Debian GNU/Linux 10 (buster)
> Release: 10
> Codename: buster
> 
> redis-server: *** 5:5.0.3-4+deb10u1 500

Some here. Debian 10 using rspamd from rspamd repo.

**Comment by t0rzz** [comment]
Same behavior. rspamd 2.5 from rspamd repo, not from Debian repo.

**Comment by degoya** [comment]
same error here on Debian 10 with rspamd 2.5 from the debian repo

**Comment by t0rzz** [comment]
Anyone willing/able to provide a fix/explain the error? And why did it start happening with rspam v2.5 (rspam repo)? This bug so far makes the whole project useless, as spam mails are coming without rspam learning if it’s a spam mail.

**Comment by andybalholm** [comment]
I'm pretty sure that it only happens when an action is "forced" on the message. If the message completes the normal scoring process, and the action is assigned based on the score, auto-learn proceeds as normal. I think the behavior is actually an improvement over what it was before. (The old behavior was to auto-learn all messages with "forced" actions, regardless of what the bayes score would have been if the bayes classifier had run on them. Auto-learn is much more effective if it only trains the filter on its mistakes, not on all messages, so auto-learning all forced messages was not ideal.) So the only real bug is the misleading error message.

If you want to get your auto-learning back, you will need to use fewer forced actions, and just make the scores on those rules high enough that they won't be overriden by any other rules.

**Comment by t0rzz** [comment]
I’ve been stuck with 1.7k mails “learned” since this bug appeared, so learning has totally stopped working. Even if the score is enough to perform an action, it will throw the error. Also I’m using the default values for the scoring system. If I run the rspam learn_spam command manually, it learns the new mails I put in the Spam folder (increasing that 1.7k learned mails, obviously). Something somewhere is broken, and I can’t seem to be able to fix it.

**Comment by andybalholm** [comment]
Are the messages that aren't being learned getting scanned by the bayes classifier? If so, most of them will have either the BAYES_SPAM or BAYES_HAM symbol.

**Comment by diolwe** [comment]
Ubuntu 18.04 with rspamd 2.5-156~bionic: Ran into the same problem today. I was able to revive rspamd by feeding it spam on the command line via

` find /home/username/Maildir/.Junk/cur/ -type f -exec /usr/bin/rspamc learn_spam {} \;`

Problem with

`rspamd_task_process: learn error: Unknown statistics error, found when storing data on backend; classifier: bayes`

appeared and localhost:8080 stopped showing entries on history-tab after changing rows-per-page settings. Restarting rspamd did not help; reboot did not help either.
